### ISSUE TITLE
Building the runtime with -O0 rather than -O2 causes testsuite to fail

### ISSUE THREAD

**Issue by sadiqj** [ocamllabs/ocaml-multicore#522]
PR: https://github.com/ocaml-multicore/ocaml-multicore/pull/521

Failure: https://github.com/sadiqj/ocaml-multicore/runs/2230302080

```
List of unexpected errors:
    tests/lib-systhreads/'test_c_thread_register.ml' with 1.1.1.1 (bytecode) 
    tests/lib-systhreads/'test_c_thread_register.ml' with 1.1.1.2 (native) 
    tests/parallel/'test_c_thread_register.ml' with 1.2 (native) 
    tests/parallel/'test_c_thread_register.ml' with 1.1 (bytecode) ```

**Comment by ctk21** [comment]
We actually don't need to hack the script to get the behaviour we want.

```sh
./configure CFLAGS='-O0' && make -j
```
will build a compiler with `-O0` at the end of the arguments which will apply over the `-O2` default (at least for gcc and clang as seen on the compiler explorer)

**Comment by ctk21** [comment]
Fixed by #596
